scripts/jenkins/ardana/gerrit/gerrit.py

```diff
@@ -0,0 +1,280 @@
+            raise Exception(
+                "Unable to get a unique change for {}{}."
+                "This can also happen if the same change-id is used "
+                "in multiple gerrit projects.".format(
+                    change_id,
+                    " and branch {}".format(branch) if branch else '')
+            )
+        elif len(response_json) != 1:
+            raise Exception("Unable to find a change for {}{}".format(
+                change_id, " and branch {}".format(branch) if branch else ''))
+
+        self._set_change_object(response_json[0])
+
+    def _set_change_object(self, change_object):
+        self._change_object = change_object
+
+    def _load_change_object(self, patchset=None):
+        self.id = str(self._change_object['_number'])
+        self.change_id = self._change_object['change_id']
+        self.gerrit_project = self._change_object['project'].split('/')[1]
+        self.status = self._change_object['status']
+        if not patchset:
+            current_revision = self._change_object['current_revision']
+            revision_obj = self._change_object['revisions'][current_revision]
+        else:
+            revisions = [r for r in self._change_object['revisions'].values()
+                         if r['_number'] == int(patchset)]
+            if len(revisions) != 1:
+                raise Exception(
+                    "Unable to find patchset {} for change {} ".format(
+                        patchset, self.id))
+            revision_obj = revisions[0]
+        self.patchset = str(revision_obj['_number'])
+        self.url = revision_obj['fetch']['anonymous http']['url']
+        self.ref = revision_obj['fetch']['anonymous http']['ref']
+        self.branch = self._change_object['branch']
+        self.subject = revision_obj['commit']['subject']
+        self.commit_message = revision_obj['commit']['message']
+        self.mergeable = self._change_object.get('mergeable', False)
+        self.submittable = self._change_object.get('submittable', False)
+
+    def _find_dependency_headers(self):
+        # Search for Depends-On headers
+        dependencies = []
+
+        for match in DEPENDS_ON_RE.findall(self.commit_message):
+            # Grab out the change-id
+            if match[6]:
+                # Group 6 is the change-id matcher from the URL
+                # NOTE: Because we only pull out the change-id here,
+                #       if a patchset was included in the URL it will be
+                #       ignored at this point.
+                #       The check later on will be missed and thus no warning
+                #       is raised.
+                dependencies.append(match[6])
+            elif match[7]:
+                # Group 7 is the gerrit ID in format Ia32...234a (41 chars)
+                dependencies.append(match[7])
+        # Reverse the order so that we process the oldest Depends-On first
+        dependencies.reverse()
+        return dependencies
```

Comment:
We may want to cache this particular check. Granted it'd only be saving on a few extra regex checks, but it could be helpful as both get_dependencies() and depends_on_directly() use it.